gcsfs/core.py

```diff
@@ -1447,8 +1460,11 @@ def _fetch_range(obj_dict, session, start=None, end=None):
         head = {'Range': 'bytes=%i-%i' % (start, end - 1)}
     else:
         head = None
-    back = session.get(obj_dict['mediaLink'], headers=head)
-    data = back.content
+    r = session.get(obj_dict['mediaLink'], headers=head)
+    r.raise_for_status()
+    md = b64decode(r.headers['X-Goog-Hash'].split('md5=')[1])
+    assert md5(r.content).digest() == md, "Checksum failure"
```

Comment:
Could factor out checksum verification code used here and in cat.

Reply:
Ah, actually I don't think this should be here - you don't expect to get any hash when downloading only part of the file.